### PR TITLE
Rebild Main structure of programm:

### PR DESCRIPTION
**Pull request by 8Invader8**
-Make Menu;
-Make FirstOperation(hide inner operetions);
-Make Class Constants;
and another small changes.

### CODE REVIEW

File: src/Constants.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import java.util.Scanner;
+
+public class Constants {
+    public static final Scanner scanner = new Scanner(System.in);
+    public static final String LANGUAGECODE = "абвгґдеєжзиіїйклмнопрстуфхцчшщьюя .,\":-!?";
+    public static final String MENU_INFO =
```
Comment (Volodymyr95) [inline src/Constants.java]: use Java text blocks

File: src/Constants.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import java.util.Scanner;
+
+public class Constants {
+    public static final Scanner scanner = new Scanner(System.in);
```
Comment (Volodymyr95) [inline src/Constants.java]: don't create it like constant, create a new one where u need it

File: src/Constants.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import java.util.Scanner;
+
+public class Constants {
```
Comment (Volodymyr95) [inline src/Constants.java]: make it final and add private constructor

File: src/Encoding.java [hunk 4]
```diff
@@ -5,9 +5,8 @@
 
 import static java.nio.file.Files.writeString;
 
-public class Encoding {
+public class Encoding extends Constants{
```
Comment (Volodymyr95) [inline src/Encoding.java]: don't extends constant

File: src/FirstOperation.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.readAllLines;
+
+public class FirstOperation extends Constants{
+
+    public List<String> ChooseTheFile(){
```
Comment (Volodymyr95) [inline src/FirstOperation.java]: methods name should start from lower case letter

File: src/Menu.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+public class Menu extends Constants{
+
+    public void Run(){
+        System.out.print(MENU_INFO);
+        String in = scanner.nextLine();
+        List<String> list ;
+
+            //Вибір напрямку програми
+            if (in.equals("1")) {
```
Comment (Volodymyr95) [inline src/Menu.java]: "1","2" and other numbers should be like constants:
private(or public) static final String ENCODING_ITEM = "1";

File: src/AnalyzorForCrypt.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        char keyCharPersent = '%';
+        for (int i = 1; i < 4; i++) {
+            if (Objects.equals(keyCharStar, copyOfList[i])){
+                count++;
+            } else if (Objects.equals(keyCharPersent, copyOfList[i])) {
+                count += 2;
+
+            }
+        }
+        if (count == 1) {
```
Comment (Volodymyr95) [inline src/AnalyzorForCrypt.java]: 1,2, are magic numbers

File: src/AnalyzorForCrypt.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    public int analyzorForCrypt(String list){
+        char[] copyOfList = list.toCharArray();
+        int count = 0;
+        char keyCharStar = '*';
+        char keyCharPersent = '%';
+        for (int i = 1; i < 4; i++) {
+            if (Objects.equals(keyCharStar, copyOfList[i])){
+                count++;
+            } else if (Objects.equals(keyCharPersent, copyOfList[i])) {
```
Comment (Volodymyr95) [inline src/AnalyzorForCrypt.java]: not needed equals for char

File: src/BruteForce.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+public class BruteForce {
+
+    public BruteForce(String list){
+        int number = new AnalyzorForCrypt().analyzorForCrypt(list);
+        switch(number){
+            case 3 -> new KeyOne(list);
```
Comment (Volodymyr95) [inline src/BruteForce.java]: 3,5,7,9 are magic numbers

File: src/Decoding.java [hunk 10]
```diff
@@ -1,40 +1,23 @@
...
-                    }else if(j - numb < 0){
-                        int count = j - numb + LANGUAGECODE.length() ;
-                        result[i] = LANGUAGECODE.charAt(count);
-                    }else {
-                        result[i] = LANGUAGECODE.charAt(j - numb);
-                    }
-                }
+
+        for (char c : copyOfList) {
+            // (char)(c - number);
```
Comment (Volodymyr95) [inline src/Decoding.java]: comented code

File: src/Encoding.java [hunk 11]
```diff
@@ -1,44 +1,48 @@
...
-import static java.nio.file.Files.writeString;
 
 public class Encoding {
 
-    public static final String LANGUAGECODE = "абвгґдеєжзиіїйклмнопрстуфхцчшщьюя .,\":-!?";
-    public Encoding(String code,int numb){
-        char[] result = code.toCharArray();
+    public Encoding(String list, int number){
+        char[] copyOfList = list.toCharArray();
         Path path = Path.of("d://result.txt");
```
Comment (Volodymyr95) [inline src/Encoding.java]: hard-coded url, it should be passed from console(user input)

File: src/FirstOperation.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+    private Scanner scanner = new Scanner(System.in);
+    public List<String> chooseTheFile(){
+        System.out.println("Введіть шлях до файлу: ");
+        String filePath = scanner.nextLine();
+        Path path = Path.of(filePath);
+        try {
+            if (exists(path)) {
+                List<String> list = readAllLines(path);
+                return list;
```
Comment (Volodymyr95) [inline src/FirstOperation.java]: return readAllLines(path);

File: src/FirstOperation.java [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.Scanner;
+
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.readAllLines;
+
+public class FirstOperation {
+
+    private Scanner scanner = new Scanner(System.in);
+    public List<String> chooseTheFile(){
+        System.out.println("Введіть шлях до файлу: ");
```
Comment (Volodymyr95) [inline src/FirstOperation.java]: use English

File: src/KeyFour.java [hunk 14]
```diff
@@ -0,0 +1,5 @@
+public class KeyFour {
+    public KeyFour(String list){
+        new Decoding(list, 9);
```
Comment (Volodymyr95) [inline src/KeyFour.java]: 9 is magic number

File: src/Menu.java [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    public static final String BRUTE_FORCE_ITEM = "3";
+    public static final String EXIT_ITEM = "0";
+
+    public void Run(){
+        System.out.print(MENU_INFO);
+        Scanner menuIn = new Scanner(System.in);
+        String menuChooseItem = menuIn.nextLine();
+        List<String> list ;
+
+            //Вибір напрямку програми
```
Comment (Volodymyr95) [inline src/Menu.java]: delete comments

File: src/AnalyzorForCrypt.java [hunk 16]
```diff
@@ -1,28 +1,32 @@
-import java.util.Objects;
-
-public class AnalyzorForCrypt {
 
+public class AnalyzorForCrypt extends Encoding{
+    private int count = 0;
```
Comment (Volodymyr95) [inline src/AnalyzorForCrypt.java]: make it constants

File: src/AnalyzorForCrypt.java [hunk 17]
```diff
@@ -1,28 +1,32 @@
...
-        char keyCharStar = '*';
-        char keyCharPersent = '%';
+
         for (int i = 1; i < 4; i++) {
-            if (Objects.equals(keyCharStar, copyOfList[i])){
+            if (copyOfList[i] == keyCharStar){
                 count++;
-            } else if (Objects.equals(keyCharPersent, copyOfList[i])) {
+            }else if (copyOfList[i] == keyCharPersent) {
                 count += 2;
```
Comment (Volodymyr95) [inline src/AnalyzorForCrypt.java]: 2 is magic number

File: src/BruteForce.java [hunk 18]
```diff
@@ -1,14 +1,15 @@
...
+    private int stepByNine = 9;
     public BruteForce(String list){
         int number = new AnalyzorForCrypt().analyzorForCrypt(list);
-        switch(number){
-            case 3 -> new KeyOne(list);
-            case 5 -> new KeyTwo(list);
-            case 7 -> new KeyTree(list);
-            case 9 -> new KeyFour(list);
-            default -> System.out.println("I don`t know how to help you!");
+        if(number == stepByThree){ new KeyOne(list);
```
Comment (Volodymyr95) [inline src/BruteForce.java]: u can refactor it using map

Comment (8Invader8) [reply src/BruteForce.java]: I was try to remake "if" to "HashMap", but in final rezult, I`ll have worst and bigger code than I had.
Or I don`t understand correct what I should do

File: src/AnalyzorForCrypt.java [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+    private int count = 0;
+    private static final int KEY_ONE = 1;
+    private static final int KEY_TWO = 2;
+    private static final int KEY_THREE = 3;
+    private static final int KEY_FOUR = 4;
+    private static final int BIG_NUMBER_FOR_EXAMPLE = 9;
+    private static final char KEY_CHAR_STAR = '*';
+    private static final char KEY_CHAR_PERSENT = '%';
+
+    public int analyzorForCrypt(String list){
```
Comment (Volodymyr95) [inline src/AnalyzorForCrypt.java]: list, is not informative name for String

File: src/AnalyzorForCrypt.java [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+        char[] copyOfList = list.toCharArray();
+
+        for (int i = 1; i < 5; i++) {
```
Comment (Volodymyr95) [inline src/AnalyzorForCrypt.java]: why 5? 5 is magic number

File: src/Decoding.java [hunk 21]
```diff
@@ -1,40 +1,24 @@
 import java.io.IOException;
+
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
-import java.util.Objects;
-
-import static java.nio.file.Files.writeString;
 
 public class Decoding extends Encoding{
```
Comment (Volodymyr95) [inline src/Decoding.java]: Decoding is not Encoding, inheritance you can use for 'is a' relationship

File: src/Encoding.java [hunk 22]
```diff
@@ -1,44 +1,58 @@
...
+            try {
+                Files.writeString(path, String.valueOf((char)(c + number)), StandardOpenOption.APPEND);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
         }
     }
 
     public Encoding() {
+
```
Comment (Volodymyr95) [inline src/Encoding.java]: empty constructor should be first

File: src/FirstOperation.java [hunk 23]
```diff
@@ -0,0 +1,34 @@
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Scanner;
+
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.readAllLines;
+
+public class FirstOperation {
```
Comment (Volodymyr95) [inline src/FirstOperation.java]: FirstOperation is not informative name

File: src/FirstOperation.java (hunk 12 above)
Comment (Volodymyr95) [reply src/FirstOperation.java]: not fixed still

File: src/KeyFour.java [hunk 24]
```diff
@@ -0,0 +1,8 @@
+public class KeyFour extends Encoding{
+    private int keyNine = 9;
```
Comment (Volodymyr95) [inline src/KeyFour.java]: what keyNine mean?

File: src/KeyFour.java [hunk 25]
```diff
@@ -0,0 +1,8 @@
...
+    protected int stepForString = 5;
```
Comment (Volodymyr95) [inline src/KeyFour.java]: what stepForString mean?

File: src/api/Main.java [hunk 26]
```diff
@@ -0,0 +1,9 @@
+package api;
+
+public class Main {
+    public static void main(String[] args) {
+
+        new Menu().Run();
```
Comment (Volodymyr95) [inline src/api/Main.java]: why method name start from upper letter?

File: src/api/Menu.java [hunk 27]
```diff
@@ -0,0 +1,10 @@
+package api;
+
+public class Menu {
+
+    public void Run(){
+        while (MenuItem.isRunning){
+            new MenuItem();
```
Comment (Volodymyr95) [inline src/api/Menu.java]: not needed create here object of menu item, u can move code from menu item here

File: src/api/crypting/AnalyzorForCrypt.java [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+
+        for (int i = 1; i < IMPORTANT_NUMBER_FOR_CRYPT; i++) {
+            if (copyOfList[i] == KEY_CHAR_STAR){
+                count++;
+            }else if (copyOfList[i] == KEY_CHAR_PERСENT) {
+                count += KEY_TWO;
+            }
+        }
+
+        if (count == KEY_ONE) {
```
Comment (Volodymyr95) [inline src/api/crypting/AnalyzorForCrypt.java]: use switch case

File: src/api/crypting/FileAndKeyForCrypt.java [hunk 29]
```diff
@@ -0,0 +1,34 @@
...
+        try {
+            if (exists(path)) {
+                return readAllLines(path);
+            }
+        } catch (FileNotFoundException e){
+            System.out.println("Cannot find file" + e);
+        } catch (Exception e) {
+        System.out.println("Oh, you have to a Big one. So, do something!" + e);
+        }
+        return null;
```
Comment (Volodymyr95) [inline src/api/crypting/FileAndKeyForCrypt.java]: don't return null it's a bad practice

File: src/api/crypting/SecretKey.java [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    public SecretKey(int keyForCrypt){
+        if(keyForCrypt == firstKeyForStep){
+            try {
+                Files.writeString(Path.of(scanner.nextLine())," * ");
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+
+        } else if (keyForCrypt == secondKeyForStep) {
```
Comment (Volodymyr95) [inline src/api/crypting/SecretKey.java]: don't use == for comparing

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**